typebinder/src/contexts/exporter.rs

```diff
@@ -170,25 +171,56 @@ impl ExporterContext<'_> {
         generics: &Generics,
         fields: Vec<Field>,
     ) -> Result<Solved<Vec<ExportStatement>>, TsExportError> {
-        let mut imports = Vec::new();
-        let mut constraints = GenericConstraints::default();
-        let members: Vec<TypeMember> = fields
+        #[derive(Default)]
+        struct Accumulator {
+            members: Vec<TypeMember>,
+            extends_clause: Option<InterfaceExtendsClause>,
+            imports: Vec<ImportEntry>,
+            constraints: GenericConstraints,
+        }
+
+        let Accumulator {
+            members,
+            extends_clause,
+            imports,
+            constraints,
+        } = fields
             .into_iter()
-            .filter_map(|field| {
-                if field.attrs.skip_serializing() {
-                    return None;
+            .try_fold(Accumulator::default(), |mut acc, field| {
+                let Accumulator {
+                    members,
+                    extends_clause,
+                    imports,
+                    constraints,
+                } = &mut acc;
+                if !field.attrs.skip_serializing() {
+                    let solver_info = MemberInfo::from_generics_and_field(generics, &field);
+                    let mut solved = self.solve_member(&solver_info)?;
+                    imports.append(&mut solved.import_entries);
+                    constraints.merge(solved.generic_constraints);
+                    if field.attrs.flatten() {
+                        let TypeMember::PropertySignature(PropertySignature { inner_type, .. }) =
```

Comment:
IDK how I feel about let-else as opposed to a clear `match` block here. It would also clearly mark what types are ignored here, and I think it's important here

Reply:
I guess you are talking about the next instruction? I think that's pretty heavy to use a match where a let-else could work, but whatever.